chaoslib/litmus/disk_fill_by_litmus.yml

```diff
@@ -0,0 +1,161 @@
+  args:
+    executable: /bin/bash
+  register: pod_name
+
+- name: Get ephemeral-storage Limit Value
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.spec.containers[?(@.name=="{{ app_container }}")].resources.limits.ephemeral-storage}' | grep -Eo '+|[0-9]+'
+  args:
+    executable: /bin/bash
+  register: limit_value
+
+- name: Get ephemeral-storage Limit storage unit
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.spec.containers[?(@.name=="{{ app_container }}")].resources.limits.ephemeral-storage}' | grep -Eo '[[:alpha:]]+'
+  args:
+    executable: /bin/bash
+  register: limit_value_storage
+
+- name: Get ephemeral-storage Request Value
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.spec.containers[?(@.name=="{{ app_container }}")].resources.requests.ephemeral-storage}' | grep -Eo '+|[0-9]+'
+  args:
+    executable: /bin/bash
+  register: req_value
+
+- name: Get ephemeral-storage Request storage unit
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.spec.containers[?(@.name=="{{ app_container }}")].resources.requests.ephemeral-storage}' | grep -Eo '[[:alpha:]]+'
+  args:
+    executable: /bin/bash
+  register: req_value_storage
+
+- name:
+  include_vars: /experiments/generic/disk_fill/test_vars_diskfill.yml
+
+- name: Fetch the value of Request Storage Unit in KB
+  set_fact:
+    req_storage_KB: "{{ lookup('vars', req_value_storage.stdout) }}"
+
+- name: Fetch the value of Limit Storage Unit in KB
+  set_fact:
+    limit_storage_KB: "{{ lookup('vars', limit_value_storage.stdout)}}"
+
+- name: Multiply the request value with unit in KB
+  set_fact:
+    req_KB: "{{ req_value.stdout|int * req_storage_KB|int }}"
+
+- name: Multiply the limit value with unit in KB
+  set_fact:
+    limit_KB: "{{ limit_value.stdout|int * limit_storage_KB|int }}"
+
+- name: Convert the fill percent
+  set_fact: 
+    fill: "{{ fill_percent |float / 100 }}"
+
+- name: Fill percent into storage
+  set_fact:
+    to_fill: "{{ fill|float * limit_KB|float }}"
+
+- name: Get the Container ID
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.status.containerStatuses[?(@.name=="{{ app_container }}")].containerID}' | awk -F'//' '{print $2}'
+  args:
+    executable: /bin/bash
+  register: containerID
+
+- name: Identify the application node
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} --no-headers -o custom-columns=:spec.nodeName
+  args:
+    executable: /bin/bash
+  register: app_node
+
+- name: Create the DaemonSet for Disk-Fill
+  shell: >
+    kubectl apply -f /experiments/generic/disk_fill/disk_fill_ds.yaml -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: ds_status
+  
+- name: Confirm that the disk-fill daemonSet is running on all nodes
```

Comment:
We may need to convert this to a check on "all available nodes". It may so happen one of the daemonset pods is not scheduled due to a tainted or not-ready node. The available nodes need to be determined before making this check. It needs to go into pumba-based experiments and elsewhere where such ds checks are used.